mealie/routes/households/controller_mealplan.py

```diff
@@ -40,57 +41,77 @@ def mixins(self):
             self.registered_exceptions,
         )
 
+    def _get_random_recipes_from_mealplan(
+        self, plan_date: date, entry_type: PlanEntryType, limit: int = 1
+    ) -> list[Recipe]:
+        """
+        Gets rules for a mealplan and returns a list of random recipes based on the rules.
+        May return zero recipes if no recipes match the filter criteria.
+
+        Recipes from all households are included unless the rules specify a household filter.
+        """
+
+        rules = self.repos.group_meal_plan_rules.get_rules(PlanRulesDay.from_date(plan_date), entry_type.value)
+        cross_household_recipes = get_repositories(self.session, group_id=self.group_id, household_id=None).recipes
+
+        tags: list[PlanTag] = []
+        categories: list[PlanCategory] = []
+        households: list[PlanHousehold] = []
+        for rule in rules:
+            if rule.tags:
+                tags.extend(rule.tags)
+            if rule.categories:
+                categories.extend(rule.categories)
+            if rule.households:
+                households.extend(rule.households)
+
+        if not (tags or categories or households):
+            return cross_household_recipes.get_random(limit=limit)
+
+        category_ids = [category.id for category in categories] or None
+        tag_ids = [tag.id for tag in tags] or None
+        household_ids = [household.id for household in households] or None
+
+        recipes_data = cross_household_recipes.page_all(
+            pagination=PaginationQuery(
+                page=1, per_page=limit, order_by="random", pagination_seed=self.repo._random_seed()
+            ),
+            categories=category_ids,
+            tags=tag_ids,
+            households=household_ids,
+        )
+        return recipes_data.items
+
     @router.get("/today")
     def get_todays_meals(self):
         return self.repo.get_today()
 
     @router.post("/random", response_model=ReadPlanEntry)
     def create_random_meal(self, data: CreateRandomEntry):
         """
-        create_random_meal is a route that provides the randomized functionality for mealplaners.
-        It operates by following the rules setout in the Groups mealplan settings. If not settings
-        are set, it will default return any random meal.
+        `create_random_meal` is a route that provides the randomized functionality for mealplaners.
+        It operates by following the rules set out in the household's mealplan settings. If no settings
```

Comment:
Yes, meal plan rules are at the household level. All meal plan related things are, since each household plans for meals differently. We moved them from group to household in the original households PR